build.gradle

```diff
@@ -586,3 +590,27 @@ ext.createDiffFile = { ->
 
   return file
 }
+
+private excludedClassFiles(classDirectories) {
+  // exclude classes generated from Avro
+  def generatedSources = [
```

Comment:
I tried to automatically exclude generated classes like what we did for SpotBug but this Jacoco-related tasks have differenet scope so couldn't get it work perfectly. The hardcoded-list is fine IMO.

Reply:
Hmm... this seems potentially brittle... Did you look at how the `compileAvro` task discovers all the schemas... might it be reasonable to go through the schema files with the same logic as that task (e.g. taking into account the same version overrides) and then for each file grep the first occurrence of `namespace` within it and extract the packages from there? While we're at it, we could potentially even generate a list of fully qualified names rather than just packages. I recognize it's kind of a rabbit hole.